### PR TITLE
New reporting scheme

### PR DESCRIPTION
**Pull request by pselle**
This separates the iopipe report out into a separate report per invocation, vs modifying in place.
Also updates advanced usage documentation & API

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -56,12 +56,11 @@ def handler(event, context):
 The following may be set as kwargs to the IOpipe class initializer:
 
 - client_id: your client_id (register at [www.iopipe.com](https://www.iopipe.com)
-- debug: enable debug logging.
-- url: custom URL for reporting metrics
+- debug: enable debug logging by setting this to `True`
```
Comment (ewindisch) [inline README.md]: did we kill 'url' or did we not support it? (we should support configurable urls.. it's useful for our own testing)

Comment (pselle) [inline README.md]: We updated it to regional collectors. I could add the support back if we still want it

File: README.md [hunk 2]
```diff
@@ -109,4 +109,4 @@ This makes it easy to add custom data and telemetry within your function.
 
 Provided under the Apache-2.0 license. See LICENSE for details.
 
-Copyright 2016, IOpipe, Inc.
+Copyright 2017, IOpipe, Inc.
```
Comment (ewindisch) [inline README.md]: we should leave the copyright alone or make it 2016-2017.

File: README.md [hunk 3]
```diff
@@ -72,13 +71,14 @@ from iopipe.iopipe import IOpipe
 def handler(event, context):
   iopipe = IOpipe(CLIENT_ID)
   timestamp = time.time()
+  iopipe.start_report(timestamp, context)
```
Comment (ewindisch) [inline README.md]: API wise it would seem cleaner to look more like:
`report = iopipe.Report()`
and then `report.send()`

Thoughts on this?

Comment (pselle) [inline README.md]: Yeah I could see that ... and make it so the parent module is _just_ responsible for config?

Comment (ewindisch) [inline README.md]: yep, and provides the decorator, etc.

Comment (ewindisch) [inline README.md]: Thinking more, it wouldn't be able to be `iopipe.Report()` but something more like:

```
report = iopipe.create_report(timestamp, context)
report.send()
```

Even better if it supports 'with'...

```
with iopipe.create_report(timestamp, context):
  # do stuff here
  pass
```

**Comment by pselle** [comment]
@ewindisch PR updated, please re-review

**Comment by ewindisch** [comment]
Looks good. I'd like to make some iterative improvements that might be API breaking, but I agree this isn't the patch to do that!

**Pull request merged**